model.go

```diff
@@ -366,9 +365,9 @@ func (v *Value) Apply(value reflect.Value) {
 }
 
 // ApplyDefault value to field if it is not already set.
-func (v *Value) ApplyDefault() error {
+func (v *Value) ApplyDefault(lookupEnv func(string) (string, bool)) error {
```

Comment:
This method signature can't be changed, as it would be a breaking API change.

Reply:
1. I’ve encountered difficulties using `env` in tests.

2. I was thinking about passing secrets directly from Vault instead of writing them to `env` (because writing them to `env` can lead to leaks when launching subprocesses).

It seemed logical to me, since otherwise there's a strange situation where we can pass `args` but not `env`.

I didn’t add any separate tests because I simply switched all the current `env` tests to this new method, and I’m not sure what else could be added.

Regarding API compatibility, the only option I see is to add a field to `Value` type similar to the `Mapper` field. Would that be okay?